events/renderers.py

```diff
@@ -12,3 +26,198 @@ class JSONLDRenderer(JSONRenderer):
     media_type = 'application/ld+json'
     format = 'json-ld'
     charset = 'utf-8'
+
+
+class DOCXRenderer(renderers.BaseRenderer):
+    media_type = 'application/vnd.openxmlformats-officedocument.wordprocessingml.document'
+    format = 'docx'
+    charset = None
+    render_style = 'binary'
+
+    def get_query_params(self, renderer_context, param_name):
+        return renderer_context['request'].query_params.get(param_name, '')
+
+    def get_any_language(self, dictionary, default='fi', language_codes=None):
+        if not language_codes:
+            # Default order for when a language isn't found.
+            language_codes = list(set([default, 'fi', 'sv', 'en']))
+
+        for language_code in language_codes:
+            content = dictionary.get(language_code)
+            if content:
+                return content
+        return ''
+
+    def get_date_range(self, renderer_context):
+        dates = {'start': None, 'end': None}
+
+        start_date = self.get_query_params(renderer_context, 'start')
+        if start_date:
+            dates['start'] = dateutil.parser.parse(start_date)
+
+        end_date = self.get_query_params(renderer_context, 'end')
+        if end_date:
+            dates['end'] = dateutil.parser.parse(end_date)
+
+        try:
+            days = int(self.get_query_params(renderer_context, 'days'))
+        except ValueError as e:
+            days = None
+        # Days override dates
+        if days:
+            dates['start'] = datetime.datetime.now().date()
+            dates['end'] = dates['start'] + datetime.timedelta(days=days)
+
+        return dates
+
+    def render(self, data, media_type=None, renderer_context=None):
+        language_code = getattr(settings, 'LANGUAGE_CODE', 'fi')
+        document = Document()
+
+        location = self.get_query_params(renderer_context, 'location')
+        # This could be set to something like "All locations"
+        place_name = ''
+        try:
+            place_name = Place.objects.values_list(
+                'name',
+                flat=True,
+            ).get(id=location)
+        except Place.DoesNotExist:
+            raise Http404('Location not found')
+
+        document.add_heading('Tapahtumat: %s' % place_name, 0)
+
+        # Note that these dates are for the request while the dates for
+        # individual events differ (but should be within these).
+        dates = self.get_date_range(renderer_context)
+
+        # These dates are for the entire document, not individual events.
+        # The minuses in strftime are for removing leading zeros.
+        date_string = ''
+        if dates['start']:
+            date_string += dates['start'].strftime('%-d.%-m.%Y')
+        if dates['end']:
+            date_string += '-%s' % dates['end'].strftime('%-d.%-m.%Y')
+
+        if date_string:
+            document.add_paragraph(date_string)
+
+        # We need to sort the events in chronological order, so that events
```

Comment:
Further, currently only 20 first events are printed (due to DRF pagination). Depending on DRF ordering, this results in only some of the events being listed. Default ordering field is `last_modified`, which means events may well be missing in between the printed events.

Increase the maximum number of events and force ordering of the events so that events are always ordered by starting time *before* serializing. This ensures no events are missing in the chronology.